crates/wast/src/component/import.rs

```diff
@@ -30,17 +67,175 @@ pub enum ComponentExternName<'a> {
     Kebab(&'a str),
     /// This is an interface import where the string is an ID.
     Interface(&'a str),
+    /// This is an implementation import where the string is an ID.
+    Implementation(ImplementationImport<'a>),
```

Comment:
Similar to wasm-encoder and wasmparser, I think this should have the same "flat" representation where the variants of `ImplementationImport` are inlined above. (and additionally split into `Component{Export,Import}Name`